File: code_test/acmicpc/acmicpc_10872.ts
function factorial(n: number) {
  let result = 1;
  for (let i = n; i > 0; i--) {
    result = result * i;
  }
  return result;
}

function recursive_factorial(n: number): number {
  if (n <= 1) {
    return 1;
  }

  return n * recursive_factorial(n - 1);
}

console.log(recursive_factorial(10));
// 재귀는 아직 헷갈린다!!!

File: code_test/acmicpc/acmicpc_10872.py
def factorial(n):
    result = 1
    for i in range(1, n+1):
        result = result * i
    return result


def recursive_factorial(n):
    if n <= 1:
        return 1
    return n * recursive_factorial(n-1)


print(recursive_factorial(10))
